logstash-core/src/main/java/org/logstash/execution/AbstractPipelineExt.java

```diff
@@ -190,6 +191,15 @@ public void notify(ConditionalEvaluationError err) {
             } catch (IOException ioex) {
                 LOGGER.warn("Invalid operation on closing internal resources", ioex);
             }
+
+            // if pipeline has DLQ enabled, send also there the event
+            if (javaDlqWriter != null) {
+                try {
+                    javaDlqWriter.writeEntry(err.failedEvent(), "if-statement", "if-statement", "condition evaluation error, " + lastErrorEvaluationReceived);
```

Comment:
please open an issue to refactor the writeEntry to accept a "element_type" and "element_id" instead of "plugin_type" / "plugin_id", as it doesn't make sense with the addition of the conditionals, e.g.:

```json
{
          "path" => {
        "to" => {
            "value" => "101"
        }
    },
    "@timestamp" => 2024-09-17T14:45:27.888346Z,
     "@metadata" => {
        "dead_letter_queue" => {
             "entry_time" => 2024-09-17T14:45:27.999153Z,
              "plugin_id" => "if-statement",
            "plugin_type" => "if-statement",
                 "reason" => "condition evaluation error, Unexpected conditional input combination of '<no-class>:<null-value>' (left) and 'class org.jruby.RubyFixnum:100' (right)"
        }
    },
      "@version" => "1",
          "host" => {
        "name" => "arkham.local"
    },
       "message" => "{\"path\":{\"to\":{\"value\":\"101\"}}}"
}
```

Reply:
Created #16463 to track this. Maybe the `element_type` for plugins remains the same as `plugin_type` and not `plugin_output`.
However since we are changing the format of the serialized data, could this be considered a breaking? Or at least we have to think about the fact that a future version of LS with the attributes renamed should be able to read a DLQ from a previous version. WDYT?

Reply:
we'll have to figure that out in that issue. we can either have a breaking change or add an extra field with the duplicate data and mark the others as deprecated.